### ISSUE TITLE
SentEmail.Table : Include 'Source Record ID' field to improve tracability

### ISSUE THREAD

**Issue by fvet** [microsoft/ALAppExtensions#9005]
Previously we had built our own 'Sent Emails' feature, that logged all outgoing mails, including (links to) attachments.
We're considering removing our custom feature and replace it with the upcoming 'Sent Emails' feature. 

One downside is the storage of the attachments as BLOB. We already store the attachment in separate tables, so the blob approach will duplicate the database size. (unless we forcily remove the records with 'Retention Policies') This is not inevitable.

The other thing we currently lack is a link between the initiating record and the 'Sent Emails'. We had integrated some logic to transfer the 'Source Record Id' (whenever possible, not for grouped mailing) in the 'Sent Emails' table. This allowed usefull navigation options both from the Sent Email to the source record (e.g. Sales Invoice) or to have better insights on the source record (Sales Invoice) if and what attachments have been sent by mail. (often you might want to check / debug if all attachments have been sent correctly, no attachments are missing, the PDF is correct, ...) Or in case a customer refers to a mail, you have all data on hand immediately. (This only works as long as the Sent Emails are retained)

**Ask:**
- Is it possible to improve the 'Email' framework to include a 'Source Record ID' (since it's currently not open to extend), so we can restore this logic when working with the new 'Email Feature'?

**Comment by JesperSchulz** [comment]
That is absolutely possible! By end of week, I expect us to open source the source code for our new email module and add-on apps. We would love to receive a pull request from you, which adds the capabilities you think are missing. For now, I will leave this issue open. Let's reconnect about this, once our new email solution is ready for public - then we can see, if this is something we can collaborate on, or if it is something, we'll take in as a "suggestion for improvement". No doubt, that it's a great idea.

**Comment by mazhelez** [comment]
Hey @fvet 

There has been some changes in the extensibility model of the email module. Do you mind having a second look?
Now, _Sent Email_ is extensible to maybe you can build the needed solution as an extension.

**Comment by JesperSchulz** [comment]
I am happy to inform you, that the link between the initiating record and the 'Sent Email' is being established in the upcoming major release of Business Central (version 18 in April). If the new implementation doesn't fully meet your requirements, please create a new issue for us - or even better, create a pull request for us :-)

**Comment by pri-kise** [comment]
Thanks. Yeah. @JesperSchulz  is this already available in the insider builds? I will check it out and really looking forward to this feature :)

**Comment by fvet** [comment]
@JesperSchulz I'm curious about the '**initiating record**'. What if we 'batch' mail invoices and group PDFs per customer, and thus have one mail containing several invoices, I suppose this will be natively supported ?

![image](https://user-images.githubusercontent.com/25268332/108491087-c1b48100-72a3-11eb-8bd5-6e8ade0de315.png)

**Comment by pri-kise** [comment]
@fvet  I've checked the source of BC 18 Insider Builds and multiple Records are supported by the System Application, but right now the BaseApp still needs changes to support this new feature.

@JesperSchulz should we maybe now add issues for this new feature and how to imporve it or would it be worth to create a new  Discussion as I suggested 6 weeks ago?

E.g. I don't understand why the new function AddRelation was added to the E-Mail-Codeunit and not to the message codeunit like the other functions (AddAttachment)?
